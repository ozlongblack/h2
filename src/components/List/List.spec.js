// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { List } from 'immutable';
import ListView from './List.view';

describe('List', () => {
  it('renders correctly', () => {
    const fixture: Object = {
      list: {
        data: List([
          {
            cateogry: 'category',
            description: 'description',
          },
        ]),
      },
    };

    const messages: Object = {
      'app.title': 'title',
    };

    const tree: JSON = renderer
      .create(<IntlProvider locale="en"
                            messages={messages}><ListView {...fixture} /></IntlProvider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

