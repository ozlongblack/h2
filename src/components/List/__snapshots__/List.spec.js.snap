// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <h1>
    title
  </h1>
  <button>
    Get list
  </button>
  <div>
    Loading
  </div>
</div>
`;
